File: ProjetSopraFinal/src/main/java/com/example/demo/model/Formateur.java
package com.example.demo.model;

import java.util.Set;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.OneToMany;

import com.example.demo.jsonview.JsonViews;
import com.fasterxml.jackson.annotation.JsonView;

@Entity
@DiscriminatorValue("F")
public class Formateur extends RessourceHumaine{
	
	private Set<Module> modules;
	private Set<Competence> competences;

	public Formateur() {

	}

	@JsonView(JsonViews.Common.class)
	@OneToMany (mappedBy = "formateur")
	public Set<Module> getModules() {
		return modules;
	}

	public void setModules(Set<Module> modules) {
		this.modules = modules;
	}

	@JsonView(JsonViews.Common.class)
	@OneToMany(mappedBy = "key.formateur")
	public Set<Competence> getCompetences() {
		return competences;
	}

	public void setCompetences(Set<Competence> competences) {
		this.competences = competences;
	}
	
	
	
}

File: AngularSopraFinal/src/app/model/formateur.ts
import {Module} from './module';
import {Competence} from './competence';
import {RessourceHumaine} from './ressourceHumaine';

export class Formateur extends RessourceHumaine{

  constructor(private _modules?, private _competence?) {
    super('formateur');
  }

  public get modules(): Array<Module> {
    return this._modules;
  }
  public set modules(_modules: Array<Module>) {
    this._modules = _modules;
  }

  public get competence(): Array<Competence> {
    return this._competence;
  }
  public set competence(_competence: Array<Competence>) {
    this._competence = _competence;
  }
}